### PR TITLE
Update setup

### LINKED ISSUE #41: Modernize setup
- move to a setup.cfg configuration
- update trove classifiers
- use pep517

### PR DESCRIPTION
**Pull request by CaselIT**
Fixes #41

This contains #42 so should be added merged later

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,2 +1,40 @@
...
-universal = 1
+[metadata]
+name = python-mimeparse
+version = attr: mimeparse.__version__
+description = A module provides basic functions for parsing mime-type names and matching them against a list of media-ranges.
+long_description = file: README.rst
+long_description_content_type = text/x-rst
+url = https://github.com/falconry/python-mimeparse
+author = DB Tsai
+author_email = dbtsai@dbtsai.com
```
[the remark this answers is not in the document]
Comment (vytas7) [reply setup.cfg:9]: Ideally, we should probably revert to @jcgregorio as the author, and keep us just as maintainers.
But I think we can keep DB Tsai for now (and us as maintainers), and revise if we aren't happy with how it's rendered on PyPi.

File: setup.cfg [hunk 2]
```diff
@@ -1,2 +1,40 @@
...
+    Intended Audience :: Developers
+    Topic :: Internet :: WWW/HTTP
+    Topic :: Software Development :: Libraries :: Python Modules
+keywords =
+    mime-type
+project_urls =
+    Issue Tracker=https://github.com/falconry/python-mimeparse
+
+[options]
+python_requires = >=3.6
```
Comment (vytas7) [inline setup.cfg:36]: IMHO we should keep 3.5 support as long as Falcon supports 3.5. (Even if we are vendoring.)

**Review by vytas7 — CHANGES_REQUESTED** [review]

File: setup.cfg (hunk 2 above)
Comment (CaselIT) [reply setup.cfg:36]: I guess there are no particular reason not to do that.

Comment (CaselIT) [reply setup.cfg:36]: Maybe if this is implemented https://github.com/falconry/python-mimeparse/issues/40

**Review by vytas7 — APPROVED** [review]

**Pull request merged**